### PR TITLE
Pagination improvements, choice of items per page

### PR DESCRIPTION
**Pull request by janvhs**
# Description

Add support for choosing between multiple selections of items per page when doing pagination.
This affects all tables in the code-base.

## How was this tested?

Jest and Storybook

Add a documentation PR or write that no changes are required for the documentation.
- [x] **DONE**

### CODE REVIEW

File: assets/js/lib/lists/index.js [hunk 1]
```diff
@@ -11,13 +11,15 @@ export const hasOne = (elements, list) =>
...
   );
 
-export const pages = (list) => {
-  const hasRest = Boolean(list.length % 10);
-  return hasRest ? ~~(list.length / 10) + 1 : ~~(list.length / 10);
+export const pages = (list, itemsPerPage = 10) => {
+  const hasRest = Boolean(list.length % itemsPerPage);
+  return hasRest
+    ? ~~(list.length / itemsPerPage) + 1
+    : ~~(list.length / itemsPerPage);
```
Comment (balanza) [inline assets/js/lib/lists/index.js:18]: **praise:** ‏

TIL: `~~` is `Math.floor`

File: assets/js/common/Table/Pagination.jsx [hunk 2]
```diff
@@ -1,50 +1,75 @@
...
+
+import Select from '@common/Select';
 
 const getPagesArray = (pages) => Array.from({ length: pages }, (_, i) => 1 + i);
 
-function Pagination({ pages, currentPage, onSelect }) {
+function Pagination({
+  pages,
+  currentPage,
+  onSelect,
```
Comment (dottorblaster) [inline assets/js/common/Table/Pagination.jsx:12]: Do we wanna have a noop here as well?

File: assets/js/common/Table/Pagination.jsx [hunk 3]
```diff
@@ -1,50 +1,75 @@
...
+  itemsPerPageOptions = [10],
+  onChangeItemsPerPage = noop,
+}) {
   const pagesList = getPagesArray(pages);
+
+  // Can happen if the items per page get changed, while being on the last
+  // page
+  if (currentPage > pages) {
+    onSelect(pages);
+  }
```
Comment (dottorblaster) [inline assets/js/common/Table/Pagination.jsx:23]: Good catch

File: assets/js/common/Table/Table.jsx [hunk 4]
```diff
@@ -76,6 +76,10 @@ function Table({
 
   const [filters, setFilters] = useState([]);
   const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPageOptions = [10, 20, 50, 75, 100];
```
Comment (dottorblaster) [inline assets/js/common/Table/Table.jsx:79]: I think we can have this outside the component's scope, it would work the same and the list would be declared just once

File: assets/js/common/Table/Table.stories.jsx [hunk 5]
```diff
@@ -178,7 +178,12 @@ export function Populated() {
 }
 
 export function Paginated() {
-  return <Table config={{ ...config, pagination: true }} data={data} />;
+  return (
+    <Table
+      config={{ ...config, pagination: true }}
+      data={data.concat(data).concat(data).concat(data)}
```
Comment (dottorblaster) [inline assets/js/common/Table/Table.stories.jsx:184]: wat

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just some sparse comments but I think this is really good.

I noticed this is appearing on all the tables now, do we want this to happen? Do we want to have a prop to toggle this behavior? I don't have hard feelings about the direction.

File: assets/js/lib/lists/index.js (hunk 1 above)
Comment (dottorblaster) [reply assets/js/lib/lists/index.js:18]: (I have to admit I just coded this after looking at some very dark corner on the internet)

File: assets/js/common/Table/Pagination.jsx [hunk 6]
```diff
@@ -1,50 +1,75 @@
...
-                border: pageNumber % 2 === 0,
-                'border-t': pageNumber % 2 === 1,
-                'border-b': pageNumber % 2 === 1,
-              }
-            );
+    <div className="flex justify-between p-2 bg-gray-50 width-full">
+      <div className="flex pl-3 items-center">
+        <span className="pr-2 text-gray-600">Results per page </span>
+        <Select
+          className="z-50"
```
Comment (jamie-suse) [inline assets/js/common/Table/Pagination.jsx:30]: :thinking: why did you need to adjust the Z-index?

File: assets/js/common/Table/Pagination.jsx [hunk 7]
```diff
@@ -1,50 +1,75 @@
...
-                'text-gray-600': currentPage !== pageNumber,
-                'text-jungle-green-500': currentPage === pageNumber,
-                border: pageNumber % 2 === 0,
-                'border-t': pageNumber % 2 === 1,
-                'border-b': pageNumber % 2 === 1,
-              }
-            );
+    <div className="flex justify-between p-2 bg-gray-50 width-full">
+      <div className="flex pl-3 items-center">
+        <span className="pr-2 text-gray-600">Results per page </span>
```
Comment (jamie-suse) [inline assets/js/common/Table/Pagination.jsx:28]:
```suggestion
        <span className="pr-2 text-gray-600">Results per page</span>
```

**Review by dottorblaster — APPROVED** [review]
LGTM, feel free to rebase and merge :+1:
